sdk/quota/arm-quota/samples-group-quota/SetGroupQuotaLimit.Request.js

```diff
@@ -0,0 +1,108 @@
+Use the polling API - OperationsStatus URI specified in Azure-AsyncOperation header field, with retry-after duration in seconds to check the intermediate status. This API provides the finals status with the request details and status.
+ * x-ms-original-file: specification/quota/resource-manager/Microsoft.Quota/preview/2023-06-01-preview/examples/GroupQuotaLimitsRequests/PutGroupQuotaLimitsRequests-Compute.json
+ */
+async function groupQuotaLimitsRequestsCreateOrUpdate() {
+  const managementGroupId = "testMgIdRoot";
+  const groupQuotaName = "sdk-test-group-quota";
+  const resourceProviderName = "Microsoft.Compute";
+  const resourceName = "standarddv4family";
+  const credential = new DefaultAzureCredential();
+  const client = new AzureQuotaExtensionAPI(credential);
+  const groupQuotaRequest = {
+    properties: {
+      requestedResource: {
+        properties: {
+          limit: 21,
+          region: "eastus",
+          comments: "Testing the SDK"
+        }
+      }
+    }
+  };
+  const options = {
+    groupQuotaRequest,
+  };
+  const result = await client.groupQuotaLimitsRequest.beginCreateOrUpdate(
+    managementGroupId,
+    groupQuotaName,
+    resourceProviderName,
+    resourceName,
+    options
+  );
+ 
+  const rawResponse = JSON.parse(result.toString());
+  const operationLocation = rawResponse.state.config.operationLocation;
+  const requestId = operationLocation.split('/').pop().split('?')[0];
+
+  console.log(requestId);
+  
+  const startTime = Date.now();
+  const runTime = 3 * 60 * 1000; // 3 minutes in milliseconds
+  const interval = 30 * 1000; // 30 seconds in milliseconds
+
+  while (Date.now() - startTime < runTime) {
+      // Your code to execute every 30 seconds
+      const status = await client.groupQuotaLimitsRequest.get(
+        managementGroupId,
+        groupQuotaName,
+        requestId,
+      );
+
+      var provisioningState = status?.properties?.provisioningState;
+      console.log(provisioningState);
+      var finalState = "";
+      if( provisioningState != "Accepted" && provisioningState != "InProgress"){
+
+        if( provisioningState == "Escalated"){
+          finalState = provisioningState;
+          console.info("Request has been escalated. Contact your capacity manager for next steps");
+          break;
+        }else if( provisioningState == "Failed"){
+          finalState = provisioningState;
+          console.info("Failed with FaultCode:" + status?.properties?.faultCode)
+          break;
+        }else{
+          finalState = provisioningState;
+          console.info("Request was successful");
+          break;
+        }
+      }
+
+      // Wait for 30 seconds
+      const waitUntil = Date.now() + interval;
+      while (Date.now() < waitUntil) {
+          // Busy-wait loop
+      }
+  }
+
+  if(finalState != "Succeeded" && finalState != "Failed"){
```

Comment:
I feel like this logic is quite complex for customers at least if they want to write the right code for example they need to get the location from serliazed poller.

I was wondering if how frequent this logic would be referred for customers, if it is quite often I may prefer to have it directly in beginCreateOrUpdateAndWait and then hide every details for them.

Reply:
Hi @MaryGao How can we change the "beginCreateOrUpdateAndWait " ? As far as I know this is generic function which is generate and it is not related to Quota only. 
This change here is very custom to Quota when status is "Escalated" then customer should do some manual operations.

Reply:
Yeah what I mean is if these customization should be in sample code or in our src code, as I mentioned in group chat there are two parts, the first one could be done by adjusting source code but for `escalated` state, it depends on if we need customer's awareness for this.

- PUT + Azure-AsyncOperation + Location header, this is not supported in SDK. 
- Non-standard provisioning state: Escalated, this is non-terminal state and when we get this state we need customers for manual step `Contact your capacity manager for next steps`.